lib/c_glib/src/thrift/c_glib/protocol/thrift_protocol.c

```diff
@@ -61,15 +61,15 @@ thrift_protocol_set_property (GObject *object, guint property_id,
   switch (property_id)
   {
     case PROP_THRIFT_PROTOCOL_TRANSPORT:
-      protocol->transport = g_value_get_object (value);
```

Comment:
Why you duplicate it? The underlaying transport should live as long as the multiplexed one. And must be destroyed after protocol is destroyed. Duplicating the transport may lead to object references hold and maybe memory freeing problems. 
I think this property must hold a reference to it and not a copy. 
The copy can lead to memory freeing problems.